catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java

```diff
@@ -97,31 +110,95 @@ public IcebergConfigResponse getConfig(
         .build();
   }
 
-  // TODO inject some path parameters to identify the table in a secure way to then do the
-  //  access-check against the table and eventually sign the request.
-  //  The endpoint can be tweaked sing the S3_SIGNER_ENDPOINT property, signing to be
-  //  enabled via S3_REMOTE_SIGNING_ENABLED.
   @POST
   @Path("/v1/{prefix}/s3-sign/{identifier}")
   @Blocking
-  public IcebergS3SignResponse s3sign(
+  public Uni<Response> s3sign(
       IcebergS3SignRequest request,
       @PathParam("prefix") String prefix,
-      @PathParam("identifier") String identifier) {
-    ParsedReference ref = decodePrefix(prefix).parsedReference();
+      @PathParam("identifier") String identifier)
+      throws NessieNotFoundException {
 
-    URI uri = URI.create(request.uri());
+    DecodedPrefix decodedPrefix = decodePrefix(prefix);
+    ContentKey key = ContentKey.fromPathString(identifier);
+    String requestedS3Uri = S3Utils.asS3Location(request.uri());
+
+    return collectBasePrefixes(decodedPrefix, key)
```

Comment:
What happens here if multiple prefixes match? Does the `toUni` bark in that case?

Reply:
The first one is converted to `Uni` the other ones, if any, are discarded. From the javadoc:

> any item or failure events received after the first event is dropped

Reply:
Hmm I should invert `map().toUni()` it seems.